src/modules/75-cf/components/FlagArchiving/useRestoreFlagDialog.tsx

```diff
@@ -5,42 +5,78 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
-import React from 'react'
+import React, { useMemo } from 'react'
 import { Intent } from '@harness/design-system'
+import { useModalHook } from '@harness/use-modal'
 import { UseConfirmationDialogReturn, useToaster } from '@harness/uicore'
 import { useStrings, String } from 'framework/strings'
 import { useRestoreFeatureFlag, RestoreFeatureFlagQueryParams, Feature } from 'services/cf'
 import useResponseError from '@cf/hooks/useResponseError'
 import { useConfirmAction } from '@common/hooks'
+import { GitSyncFormValues, UseGitSync } from '@cf/hooks/useGitSync'
+import { GIT_COMMIT_MESSAGES } from '@cf/constants/GitSyncConstants'
+import SaveFlagToGitModal from '../SaveFlagToGitModal/SaveFlagToGitModal'
 import css from './useRestoreFlagDialog.module.scss'
 
 export interface RestoreFlagDialogProps {
   flagData: Feature
+  gitSync: UseGitSync
   queryParams: RestoreFeatureFlagQueryParams
   onRestore: () => void
 }
 
 const useRestoreFlagDialog = ({
   flagData,
+  gitSync,
   queryParams: restoreQueryParams,
   onRestore
 }: RestoreFlagDialogProps): UseConfirmationDialogReturn['openDialog'] => {
   const { showSuccess } = useToaster()
   const { getString } = useStrings()
   const { handleResponseError } = useResponseError()
 
-  const queryParams = {
-    identifier: flagData.identifier,
-    environmentIdentifier: flagData.envProperties?.environment,
-    ...restoreQueryParams
-  }
+  const { gitSyncInitialValues, gitSyncValidationSchema } = gitSync.getGitSyncFormMeta(
+    GIT_COMMIT_MESSAGES.RESTORED_FLAG
+  )
+
+  const [showGitModal, hideGitModal] = useModalHook(() => {
+    return (
+      <SaveFlagToGitModal
+        flagName={flagData.name}
+        flagIdentifier={flagData.identifier}
+        gitSyncInitialValues={gitSyncInitialValues}
+        gitSyncValidationSchema={gitSyncValidationSchema}
+        onSubmit={handleRestore}
+        onClose={hideGitModal}
+      />
+    )
+  }, [flagData.name, flagData.identifier, gitSync])
+
+  const queryParams = useMemo(() => {
+    return {
+      identifier: flagData.identifier,
+      environmentIdentifier: flagData.envProperties?.environment,
+      ...restoreQueryParams,
+      commitMsg: ''
+    }
+  }, [flagData.envProperties?.environment, flagData.identifier, restoreQueryParams])
 
   const { mutate: restoreFeatureFlag } = useRestoreFeatureFlag({
     identifier: flagData.identifier,
     queryParams
   })
 
-  const handleRestore = async (): Promise<void> => {
+  const handleRestore = async (gitSyncFormValues?: GitSyncFormValues): Promise<void> => {
+    let commitMessage = ''
+
+    if (gitSync.isGitSyncEnabled && gitSync.isAutoCommitEnabled) {
+      commitMessage = gitSyncInitialValues.gitDetails.commitMsg
+      queryParams.commitMsg = commitMessage
+    } else {
+      commitMessage = gitSyncFormValues?.gitDetails.commitMsg || ''
+      queryParams.commitMsg = commitMessage
+    }
```

Comment:
Super minor, but you can move the `queryParams.commitMsg` assignment our of the `if`/`else`
```suggestion
    } else {
      commitMessage = gitSyncFormValues?.gitDetails.commitMsg || ''
    }
    
    queryParams.commitMsg = commitMessage
```